sdk/identity/azure-identity/tests/test_live.py

```diff
@@ -2,52 +2,115 @@
+
+
+def test_client_secret_credential(live_service_principal):
     credential = ClientSecretCredential(
-        live_identity_settings["tenant_id"],
-        live_identity_settings["client_id"],
-        live_identity_settings["client_secret"],
+        live_service_principal["tenant_id"],
+        live_service_principal["client_id"],
+        live_service_principal["client_secret"],
     )
-    token = credential.get_token(ARM_SCOPE)
-    assert token
-    assert token.token
-    assert token.expires_on
+    get_token(credential)
 
 
-def test_default_credential(live_identity_settings):
+def test_default_credential(live_service_principal):
     credential = DefaultAzureCredential()
-    token = credential.get_token(ARM_SCOPE)
-    assert token
-    assert token.token
-    assert token.expires_on
+    get_token(credential)
 
 
-def test_confidential_client_credential(live_identity_settings):
+def test_confidential_client_credential(live_service_principal):
     credential = ConfidentialClientCredential(
-        client_id=live_identity_settings["client_id"],
-        client_credential=live_identity_settings["client_secret"],
+        client_id=live_service_principal["client_id"],
+        client_credential=live_service_principal["client_secret"],
         authority=KnownAuthorities.AZURE_PUBLIC_CLOUD,
-        tenant_id=live_identity_settings["tenant_id"],
+        tenant_id=live_service_principal["tenant_id"],
     )
-    token = credential.get_token(ARM_SCOPE)
-    assert token
-    assert token.token
-    assert token.expires_on
+    get_token(credential)
+
+
+@pytest.mark.skipif("TEST_IMDS" not in os.environ, reason="To test IMDS authentication, set $TEST_IMDS with any value")
+def test_imds_credential(managed_identity_id):
+    get_token(ImdsCredential())
+    if managed_identity_id:
+        get_token(ImdsCredential(client_id=managed_identity_id))
+
+
+@pytest.mark.skipif(
+    EnvironmentVariables.MSI_ENDPOINT not in os.environ or EnvironmentVariables.MSI_SECRET in os.environ,
+    reason="Legacy MSI unavailable",
+)
+def test_msi_legacy(managed_identity_id):
+    get_token(MsiCredential())
+    if managed_identity_id:
+        get_token(ImdsCredential(client_id=managed_identity_id))
+
+
+@pytest.mark.skipif(
+    EnvironmentVariables.MSI_ENDPOINT not in os.environ or EnvironmentVariables.MSI_SECRET not in os.environ,
+    reason="App Service MSI unavailable",
+)
+def test_msi_app_service(managed_identity_id):
+    get_token(MsiCredential())
+    if managed_identity_id:
+        get_token(ImdsCredential(client_id=managed_identity_id))
+
+
+def test_username_password_auth(live_user_details):
+    credential = UsernamePasswordCredential(
+        client_id=live_user_details["client_id"],
+        username=live_user_details["username"],
+        password=live_user_details["password"],
+        tenant_id=live_user_details["tenant"],
+    )
+    get_token(credential)
+
+
+@pytest.mark.manual
+def test_device_code(prints):
```

Comment:
My first thought reading this was that prints is an unused parameter, but realized it was a fixture you wrote to skip this test.  That pattern wasn't super intuitive to me; if it's consistent with broader use or pytest practices feel free to ignore this, but a "check and skip" seems like more of a decorator than a fixture?

Reply:
Thanks, that's a good suggestion, I replaced the fixture with a marker.